packages/sdk/src/types/runtime/assets/assets-v2.md

```diff
@@ -0,0 +1,100 @@
+# Asset Unloading
+
+This proposal details the API changes required to allow developers to explicitly request
+that a given asset (material, texture, sound) be unloaded from memory on the clients,
+and unassign all references to them.
+
+Today, assets can be created individually, or created en masse by loading a container
+(glTF) for assets. This proposal does not have separate unload APIs for these two cases.
+Instead, during synchronization, the adapter will examine which assets are loaded at sync
+time, send a container load request if at least one asset from that container is still
+live, then send unloads for all the unused ones. This is processor-inefficient, but
```

Comment:
"unload the unused ones" doesn't make sense to me. I would expect that assets loaded on machine A are then also loaded on machine B during synchronization. Otherwise, what if dev had preloaded an asset but aren't using it right at this moment? It should still be ready to instantly instantiate on machine B

Reply:
That is a phrasing error. It is not *unused* assets that are unloaded here. It's *unloaded*, i.e. assets that previously had `unload` called on them.

Reply:
That is a phrasing error. It is not unused assets that are unloaded here. It's unloaded, i.e. assets that previously had `unload` called on them.

---
In reply to: [300188310](https://github.com/microsoft/mixed-reality-extension-sdk/pull/369#discussion_r300188310) [](ancestors = 300188310)